cpp/src/aztec/honk/utils/power_polynomial.hpp

```diff
@@ -0,0 +1,65 @@
+#include <polynomials/polynomial.hpp>
+#include <common/max_threads.hpp>
+
+#ifndef NO_MULTITHREADING
+#include "omp.h"
+#endif
+namespace honk {
+namespace power_polynomial {
+/**
+ * @brief Generate the power polynomial vector
+ *
+ * @details Generates a vector, where v[i]=ζⁱ
+ *
+ * @param zeta
+ * @param vector_size
+ * @return barretenberg::polynomial
+ */
+barretenberg::polynomial generate_vector(barretenberg::fr zeta, size_t vector_size)
+{
+    barretenberg::polynomial pow_vector(vector_size);
+
+    constexpr size_t usefulness_margin = 4;
+    size_t num_threads = max_threads::compute_num_threads();
+    if (vector_size < (usefulness_margin * num_threads)) {
+        num_threads = 1;
+    }
+    size_t thread_size = vector_size / num_threads;
+    if ((num_threads % thread_size) != 0) {
+        thread_size += 1;
+    }
+#ifndef NO_MULTITHREADING
+#pragma omp parallel for
+#endif
+    for (size_t i = 0; i < num_threads; i++) {
+        barretenberg::fr starting_power = zeta.pow(i * thread_size);
+        for (size_t j = 0; j < (thread_size - 1); j++) {
+            pow_vector.coefficients[i * thread_size + j] = starting_power;
```

Comment:
Consider `pow_vector[i * thread_size + j]` as this allows for bound checking in debug builds? The only problem is we run tests in release with asserts so that might slow down testing. Curious to hear your thoughts.

Reply:
Are you talking about the new polynomial? The old one only checks if the polynomial is empty

Reply:
Yes, you were. Changed